### PR TITLE
style(mypy): Enforcing typing for some modules

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Adding `mypy` enforcement for a number of small modules (starting alphabetically): 

- `superset.bin`
- `superset.commands`
- `superset.common`
- `superset.dao`
- `superset.db_engines`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @dpgaspar @villebro

### CODE REVIEW

File: superset/commands/exceptions.py [hunk 1]
```diff
@@ -25,25 +25,25 @@
 class CommandException(SupersetException):
     """ Common base class for Command exceptions. """
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         if self._exception:
-            return self._exception
-        return self
+            return repr(self._exception)
```
Comment (john-bodley) [inline superset/commands/exceptions.py]: This seemed like a bug, i.e., `__repr__` should return a `str`.

File: superset/stats_logger.py [hunk 2]
```diff
@@ -24,26 +24,26 @@
...
+    def decr(self, key: str) -> None:
         """Decrement a counter"""
         raise NotImplementedError()
 
-    def timing(self, key, value):
+    def timing(self, key, value: float) -> None:
         raise NotImplementedError()
 
-    def gauge(self, key):
+    def gauge(self, key: str, value: int) -> None:
```
Comment (john-bodley) [inline superset/stats_logger.py]: This was missing the gauge value. It's  hard to tell from the statsD code whether this should be an `int`, `float` , or `both`.

File: superset/common/tags.py [hunk 3]
```diff
@@ -14,14 +14,15 @@
...
+from sqlalchemy import Metadata
+from sqlalchemy.engine import Engine
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.sql import and_, func, functions, join, literal, select
 
 from superset.models.tags import ObjectTypes, TagTypes
 
 
-def add_types(engine, metadata):
+def add_types(engine: Engine, metadata: Metadata) -> None:
```
Comment (john-bodley) [inline superset/common/tags.py]: @dpgaspar could you verify if the `metadata` type is correct as this is FAB related.

File: superset/commands/base.py [hunk 4]
```diff
@@ -23,7 +24,7 @@ class BaseCommand(ABC):
     """
 
     @abstractmethod
-    def run(self):
+    def run(self) -> Any:
```
Comment (dpgaspar) [inline superset/commands/base.py]: What do you think about setting this to `Union[None, Model]`, will mypy complain?

**Review by dpgaspar — COMMENTED** [review]
Looks good, `Metadata` at first seems right, but will check it it may come from flask_sqlalchemy

File: superset/commands/base.py (hunk 4 above)
Comment (john-bodley) [reply superset/commands/base.py]: @dpgaspar `Union[None, Model]` is equivalent to `Optional[Model]`. I've made the change.

**Comment by john-bodley** [comment]
@dpgaspar are you ok with these changes? Note I added a couple more modules; `superset.dao` and `superset.db_engines`, since I authored the PR.

**Comment by john-bodley** [comment]
ping @dpgaspar @villebro

File: superset/dao/base.py [hunk 5]
```diff
@@ -112,7 +112,7 @@ def update(cls, model: Model, properties: Dict, commit: bool = True) -> Model:
         return model
 
     @classmethod
-    def delete(cls, model: Model, commit=True):
+    def delete(cls, model: Model, commit: bool = True) -> Model:
```
Comment (villebro) [inline superset/dao/base.py:115]: I raised this in #9418 , I think that PR will benefit from this PR.

**Review by villebro — APPROVED** [review]
LGTM, I propose merging this and rebasing #9418 after that as I believe this will impact those changes.

**Pull request merged**